### PR TITLE
Ci: jest coverage reporting

### PR DESCRIPTION
**Pull request by jnurkka**

### CODE REVIEW

**Comment by jnurkka** [comment]
@sergey-cliniserve any thoughts on this proposal for checking jest test coverage? Input appreciated :) I think for our monorepos it wont work this easily, and we might want to look into some tools that help orchestrate such things, maybe lerna can be helpful in checking which packages have changed in PR and then only running jest with --coverage for those packages? Or maybe there are even more high-level tools/github apps for doing this for monorepos.

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: 'coverage'
```
Comment (sergey-cliniserve) [inline .github/workflows/coverage.yml:1]: wouldn't it be enough to fail testing phase just [enabling `--coverage`](https://github.com/Cliniserve/util/pull/5/commits/2e2736e2fb3734e4e15acdca9ab0ece74da7f747)?

File: src/isError.test.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  it('returns false if input is not an object', () => {
+    const nonErrorException = 'error';
+    expect(isError(nonErrorException)).toBeFalsy();
+  });
+  it('returns false if input is an object that does not contain message', () => {
+    const nonErrorObject = {
+      error: 'invalid',
+    };
+    expect(isError(nonErrorObject)).toBeFalsy();
+  });
```
Comment (sergey-cliniserve) [inline src/isError.test.ts:19]:
```ts
it("doesn't crash checking null", () => {
```

just because there is a null check in `isError` implementation

**Review by sergey-cliniserve — APPROVED** [review]

**Comment by sergey-cliniserve** [comment]
> @sergey-cliniserve any thoughts on this proposal for checking jest test coverage? Input appreciated :) I think for our monorepos it wont work this easily, and we might want to look into some tools that help orchestrate such things, maybe lerna can be helpful in checking which packages have changed in PR and then only running jest with --coverage for those packages? Or maybe there are even more high-level tools/github apps for doing this for monorepos.

can't find my previous comment on it 😖

I'm unsure if we need this GitHub-action bot to post a comment to each PR with coverage report. If we force 100% coverage, then CI should fail if it's lower. And if it fails, report with the same details should be in action execution logs.
But if it doesn't fail this message will just annoy everyone by unnecessary notification. Not really much against of it just can't see benefits.

For monorepos I don't know about all-in-one solution. Lerna has `--since` argument and PR check case you can say `lerna run test --since <pr-target-branch>`.

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (jnurkka) [reply .github/workflows/coverage.yml:1]: sure, then we would only lose the nice comment with the table. 🙃 But fine for me, definitely would keep CI pipeline simpler

File: src/isError.test.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import isError from './isError';
+
+describe('isError returns whether input is a valid error object', () => {
+  it('returns true if input is an object containing message field with string value', () => {
```
[the remark this answers is not in the document]
Comment (jnurkka) [reply src/isError.test.ts:4]: thanks!

File: src/isError.test.ts (hunk 2 above)
Comment (jnurkka) [reply src/isError.test.ts:19]: Thanks!

**Pull request merged**